src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -190,7 +190,7 @@ Object getVectorValueAt(String column, int index) {
       }
     }
     if (logicalType == ColumnLogicalType.BINARY && value != null) {
-      value = ((String) value).getBytes(StandardCharsets.UTF_8);
+      value = value instanceof String ? ((String) value).getBytes(StandardCharsets.UTF_8) : value;
```

Comment:
I believe we should always use hex format for BINARY data type, see https://docs.snowflake.com/en/user-guide/binary-input-output.html. CC: @sfc-gh-lsembera for confirmation

Reply:
Internally in the SDK we should use `byte[]` for BINARY, but in EpInfo we should use hex. It is in scope of my work on SNOW-663704. Server-side change was required, which was recently deployed, and later I am going to change it also in the SDK. I talked with @sfc-gh-gdoci about it, it is fine for now, I am going to fix it later.